frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -0,0 +1,629 @@
+        ).toBeInTheDocument();
+        // dashboard item
+        expect(screen.getByText(DASHBOARD.name)).toBeInTheDocument();
+      });
+
+      describe("when last visited dashboard belongs to root", () => {
+        it("should render root collection", async () => {
+          await setup({
+            dashboard: DASHBOARD_AT_ROOT,
+            mostRecentlyViewedDashboard: DASHBOARD_AT_ROOT,
+          });
+
+          // breadcrumbs
+          expect(screen.getByText(ROOT_COLLECTION.name)).toBeInTheDocument();
+          // dashboard item
+          expect(screen.getByText(DASHBOARD_AT_ROOT.name)).toBeInTheDocument();
+        });
+      });
+
+      describe("when last visited dashboard belongs to public subcollections", () => {
+        const dashboardInPublicSubcollection = createMockDashboard({
+          id: 5,
+          name: "Dashboard in public subcollection",
+          collection: SUBCOLLECTION,
+          collection_id: SUBCOLLECTION.id as number,
+          model: "dashboard",
+        });
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
+          await setup({
+            mostRecentlyViewedDashboard: dashboardInPublicSubcollection,
+          });
+
+          // breadcrumbs
+          expect(screen.getByText(ROOT_COLLECTION.name)).toBeInTheDocument();
+          expect(screen.getByText(COLLECTION.name)).toBeInTheDocument();
+          expect(screen.getByText(SUBCOLLECTION.name)).toBeInTheDocument();
+
+          // dashboard item
+          expect(
+            screen.getByText(dashboardInPublicSubcollection.name),
+          ).toBeInTheDocument();
+        });
+
+        it("should show the root collection if the question is in a personal collection", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            mostRecentlyViewedDashboard: dashboardInPublicSubcollection,
+          });
+
+          // breadcrumbs
+          expect(screen.getByText(ROOT_COLLECTION.name)).toBeInTheDocument();
+
+          // Showing personal collection means we're viewing the root collection
+          expect(
+            screen.getByRole("heading", { name: PERSONAL_COLLECTION.name }),
+          ).toBeInTheDocument();
+        });
+      });
+
+      describe("when last visited dashboard belongs to personal subcollections", () => {
+        const dashboardInPersonalSubcollection = createMockDashboard({
+          id: 5,
+          name: "Dashboard in personal subcollection",
+          collection: PERSONAL_SUBCOLLECTION,
+          collection_id: PERSONAL_SUBCOLLECTION.id as number,
+          model: "dashboard",
+        });
+
+        it("should show most recently visited dashboard if the question is in a public collection", async () => {
+          await setup({
+            mostRecentlyViewedDashboard: dashboardInPersonalSubcollection,
+          });
+
+          // breadcrumbs
```

Comment:
Maybe it would make sense to have some sort of `expectBreadcrumbs` method considering you're doing the same checks throughout the tests - something like:

```
function expectBreadcrumbs({breadcrumbs}: {breadcrumbs: string[]}) {
    for (const breadcrumb in breadcrumbs) {
        expect(
            screen.getByText(breadcrumb),
          ).toBeInTheDocument();
    }
   ... or maybe even check for order as well 🙂 
}
```

Reply:
Good suggestion. Thanks!

Reply:
Fixed in https://github.com/metabase/metabase/pull/34833/commits/ed05becadefba6ebf6fcce6b1052bfd54d364b18